### PR TITLE
Fix theming bugs

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #296, #333, #334 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [x] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Fix theming bugs and enhance the `<auro-input>` component by adding an `autocomplete` attribute. Refactor component registration to use `RuntimeUtils` and update SCSS variables for better theming consistency. Update documentation and remove deprecated scripts.

New Features:
- Introduce the `autocomplete` attribute to the `<auro-input>` component, allowing users to enable or disable autocomplete functionality with various HTML5 input standards.

Bug Fixes:
- Fix theming issues by updating SCSS variables and ensuring consistent color and padding definitions across components.

Enhancements:
- Refactor the component registration process to use `RuntimeUtils` for registering custom elements, improving modularity and reducing potential conflicts.
- Update the import paths for design tokens and adjust SCSS variable usage to align with the latest design system standards.

Build:
- Add an ESLint configuration file to standardize code quality checks across the project.

Documentation:
- Update documentation to include details about the new `autocomplete` attribute and its usage in the `<auro-input>` component.

Chores:
- Remove deprecated Husky scripts and update dependencies to their latest versions in the README and demo files.

### CODE REVIEW

File: src/auro-input.js [hunk 1]
```diff
@@ -120,6 +119,7 @@ export class AuroInput extends BaseInput {
             spellcheck="${ifDefined(this.spellcheck ? this.spellcheck : undefined)}"
             autocorrect="${ifDefined(this.autocorrect ? this.autocorrect : undefined)}"
             autocapitalize="${ifDefined(this.autocapitalize ? this.autocapitalize : undefined)}"
+            autocomplete="${ifDefined(this.autocomplete ? this.autocomplete : undefined)}"
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply src/auro-input.js:122]: Not implementing at this time.

File: demo/index.md [hunk 2]
```diff
@@ -226,8 +226,10 @@ There are two important parts of every Auro component. The <a href="https://deve
 To protect from versioning conflicts with other instances of the component being loaded, it is recommended to use our `registerComponent(name)` method and pass in a unique name.
 
 ```js
-import './node_modules/@aurodesignsystem/auro-input';
-registerComponent('custom-input');
+import { AuroInput } from './src/auro-input.js';
+import * as RuntimeUtils from '@aurodesignsystem/auro-library/scripts/utils/runtimeUtils.mjs';
+
+RuntimeUtils.default.prototype.registerComponent('custom-input', AuroInput);
 ```
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply demo/index.md:232]: This is highlighted in documentation.

File: demo/api.md [hunk 3]
```diff
@@ -416,6 +419,28 @@ The `<auro-input>` component supports setting a custom validity message specific
...
+  <span slot="trigger">See code</span>
+<!-- AURO-GENERATED-CONTENT:START (CODE:src=./../../apiExamples/autocomplete.html) -->
+<!-- The below code snippet is automatically added from ./../../apiExamples/autocomplete.html -->
+
+```html
+<auro-input autocomplete="off"></auro-input>
+<auro-input bordered autocomplete="given-name"></auro-input>
+```
+<!-- AURO-GENERATED-CONTENT:END -->
+</auro-accordion>
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply demo/api.md:442]: Closing tag is NOT missing.

File: demo/api.md [hunk 4]
```diff
@@ -33,13 +33,16 @@
...
 | [error](#error)                           | `error`                           | `String`  |                            | When defined, sets persistent validity to `customError` and sets `setCustomValidity` = attribute value. |
 | [errorMessage](#errorMessage)                    | `errorMessage`                    | `String`  |                            | Contains the help text message for the current validity error. |
 | [helpText](#helpText)                        | `helpText`                        | `String`  |                            | Deprecated, see `slot`.                          |
 | [icon](#icon)                            | `icon`                            | `Boolean` | false                      | If set, will render an icon inside the input to the left of the value. Support is limited to auro-input instances with credit card format. |
+| [iconTag](#iconTag)                         |                                   | `string`  |                            |                                                  |
```
Comment (jason-capsule42) [inline demo/api.md:45]: This should be private

File: demo/api.md [hunk 5]
```diff
@@ -33,13 +33,16 @@
...
 | [autocapitalize](#autocapitalize)                  | `autocapitalize`                  | `String`  |                            | An enumerated attribute that controls whether and how text input is automatically capitalized as it is entered/edited by the user. [off/none, on/sentences, words, characters] |
+| [autocomplete](#autocomplete)                    | `autocomplete`                    | `String`  |                            | An enumerated attribute that defines what the user agent can suggest for autocompletion. See https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete for all possible values. |
 | [autocorrect](#autocorrect)                     | `autocorrect`                     | `String`  |                            | When set to `off`, stops iOS from auto correcting words when typed into a text box. |
+| [buttonTag](#buttonTag)                       |                                   | `string`  |                            |                                                  |
```
Comment (jason-capsule42) [inline demo/api.md:38]: This should be private

File: demo/api.md [hunk 6]
```diff
@@ -33,13 +33,16 @@
 |-----------------------------------|-----------------------------------|-----------|----------------------------|--------------------------------------------------|
 | [activeLabel](#activeLabel)                     | `activeLabel`                     | `Boolean` | false                      | If set, the label will remain fixed in the active position. |
 | [autocapitalize](#autocapitalize)                  | `autocapitalize`                  | `String`  |                            | An enumerated attribute that controls whether and how text input is automatically capitalized as it is entered/edited by the user. [off/none, on/sentences, words, characters] |
+| [autocomplete](#autocomplete)                    | `autocomplete`                    | `String`  |                            | An enumerated attribute that defines what the user agent can suggest for autocompletion. See https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete for all possible values. |
```
Comment (jordanjones243) [inline demo/api.md:36]: Move link down to example. Use `autofill` word.

File: docs/partials/api.md [hunk 7]
```diff
@@ -212,6 +212,23 @@ The `<auro-input>` component supports setting a custom validity message specific
 
 </auro-accordion>
 
+### Autocomplete <a name="autocomplete"></a>
+
+Use the `autocomplete` attribute to enable or disable autocomplete functionality. The `autocomplete` attribute can be set to `on`, `off` or can be set to a specified value from HTML5 input [autocomplete](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete) standards.
```
Comment (jordanjones243) [inline docs/partials/api.md:217]: Use `auro-hyperlink`.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**